.travis.yml

```diff
@@ -60,14 +12,90 @@ before_install:
   - chmod +x docker-compose
   - sudo mv docker-compose /usr/local/bin
 
-before_script:
-  - if [ "$CC" == "gcc" ]; then export CC=gcc-4.8; fi
-  - if [ "$CXX" == "g++" ]; then export CXX=g++-4.8; fi
-  - if [ "$CC" == "clang" ]; then export CC=clang-3.8; fi
-  - if [ "$CXX" == "clang++" ]; then export CXX=clang++-3.8; fi
-  - docker-compose build base
-  - if [ "$TARGET" == "android" ]; then docker-compose build java; fi
-  - docker-compose build ${TARGET}
-
-script:
-  - docker-compose run ${TARGET}
+jobs:
+  fast_finish: true
+  allow_failures:
+    - env: TARGET=sonar-scan
```

Comment:
Yep, nicely summarized. And I think it's rather easy to achieve by just moving the variable checks inside the `if` branches (turning the check into a function to avoid code duplication). And to avoid hardcoding the "SLUG" name check, we can simply skip the scan, but not fail the build if variables are not set, assuming that if someone needs a scan, they'd define variables and those who need a scan, but forgot to set variables, would be attentive enough to check logs and see that the scan is actually skipped.